syntax = "proto3";
package connection_request;

message AddressInfo {
    string host = 1;
    uint32 port = 2;
}

enum ReadFromReplicaStrategy {
    AlwaysFromPrimary = 0;
    RoundRobin = 1;
    LowestLatency = 2;
    AZAffinity = 3;
}

enum TlsMode {
    NoTls = 0;
    SecureTls = 1;
    InsecureTls = 2;
}

message AuthenticationInfo {
    string password = 1;
    string username = 2;
}

// IMPORTANT - if you add fields here, you probably need to add them also in client/mod.rs:`sanitized_request_string`.
message ConnectionRequest {
    repeated AddressInfo addresses = 1;
    TlsMode tls_mode = 2;
    bool cluster_mode_enabled = 3;
    uint32 response_timeout = 4;
    uint32 client_creation_timeout = 5;
    ReadFromReplicaStrategy read_from_replica_strategy = 6;
    ConnectionRetryStrategy connection_retry_strategy = 7;
    AuthenticationInfo authentication_info = 8;
    uint32 database_id = 9;
}

message ConnectionRetryStrategy {
    uint32 number_of_retries = 1;
    uint32 factor = 2;
    uint32 exponent_base = 3;
}
